### ISSUE TITLE
`Simplified view` not appearing on the first page load - Follow up to #42087

### ISSUE THREAD

**Issue by kjozwiak** [brave/brave-browser#42175]
### Description

As a follow up to https://github.com/brave/brave-browser/issues/42087, the `Simplified view` prompt isn't being displayed on the initial page load. Users will need to load a compatible page and then refresh it to see the prompt. At that point, the user can go into ``simplified view`.

### Steps to reproduce

Basically the same STR/Cases outlined via https://github.com/brave/brave-browser/issues/42087#issue-2635931816.

1. enable `Simplified view for webpages` via `Hamburger Menu` -> `Settings` -> `Accessibility`
2. once enabled, load `cnn.com` and open any articles
3. notice that you won't see a `Simplified view` prompt
4. refresh the same page once again and the `Simplified view` prompt will be disaplyed

### Actual result

https://github.com/user-attachments/assets/6aba7c17-ca81-4468-93df-370490bb98c9

### Expected result

Users should be see the `Simplified view` prompt as soon as they load a compatible webpage rather than requiring a refresh to see the prompt/access `simplified view` mode. 

### Reproduces how often

Easily reproduced

### Brave version

```
Brave | 1.75.4 Chromium: 131.0.6778.39 (Official Build) canary (64-bit)
--- | ---
Revision | 26b16ff324dee28788c8470ac458b2a31d27b0b7
OS | Android 15; Build/AP41.240925.009; 35; REL
```

### Device

- Brand/model: `Pixel 6`
- Android version: `Android 15`

### Channel information

- [x] release (stable)
- [x] beta
- [x] nightly

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [ ] in the latest version of Chrome

### Miscellaneous information

The above will also be in `release/stable` once https://github.com/brave/brave-browser/issues/42087#issue-2635931816 hits via `1.73.x`.
